lib/omnibus/util.rb

```diff
@@ -65,18 +99,56 @@ def shellout(*args)
         options[:environment] = options.fetch(:environment, {}).merge(options[:env])
       end
 
+      # Double check that we don't have conflicting PATH definitions.
+      path_keys = options[:environment].keys.grep(/\Apath\Z/i)
+      if path_keys.length > 1
+        raise InvalidValue.new("shellout", "receive environment without duplicate PATH values")
+      elsif path_keys.length == 1 && path_keys.first != path_key
+        options[:environment][path_key] = options[:environment].delete(path_keys.first)
+      end
+
+      # Try our best to get rid of the "outer" omnibus ruby and any current
+      # chef-client/chef-dk installations from the path.
+      if ENV["MSYS_TOOLCHAIN"]
```

Comment:
In other places we do `windows? && ENV["MSYS_TOOLCHAIN"]`.  Any reason to exclude the windows? check here?  Perhaps we could have a use_msys_toolchain? function that does both of those checks always.